config/config.yml

```diff
@@ -87,6 +87,66 @@ ldap:
     enabled: false
     attr: ""
 
+# OAuth support.
+oauth:
+  # If enabled, users can authenticate with their Google Account.
+  google_oauth2:
+    enabled: false
+    # Credentials. Details on https://developers.google.com/identity/protocols/OpenIDConnect
+    id: ""
+    secret: ""
+    # If a domain (e.g. mycompany.com) is set, then only signups with email from this domain are allowed.
+    domain: ""
+    options:
+      # G Suite domain. If set, then only members of the domain can sign in/up.
+      # If it's empty then any google users con sign in/up.
+      hd: ""
+
+  # OpenID authentication support. If enabled, then users can authenticate with OpenID/Connect
+  open_id:
+    enabled: false
+    # Optional. If identifier set then user redirect to the OpenID provider.
+    # If not, then user is asked for identifier before redirect.
+    # Example https://openid.stackexchange.com
+    identifier: ""
+    # If a domain (e.g. mycompany.com) is set, then only signups with email from this domain are allowed.
+    domain: ""
+
+  # Github authentication support.
+  github:
+    enabled: false
+    # Application credentials.
+    key: ""
```

Comment:
As @mssola mentioned a few comments ago, it would be nice to rename this to `client_id` and `client_secret` below to avoid confusion.